source/Cosmos.System2/FileSystem/Disk.cs

```diff
@@ -23,50 +24,49 @@ public List<ManagedPartition> Partitions
-                        if (MountedPartitions[i] != null)
-                        {
-                            var data = MountedPartitions[i];
-                            part.RootPath = data.RootPath;
-                            part.MountedFS = data;
-                        }
-                        converted.Add(part);
-                        i++;
-                    }
-                }
-                else
-                {
-                    var mbr = new MBR(Host);
-                    int i = 0;
-                    foreach (var item in mbr.Partitions)
-                    {
-                        var part = new ManagedPartition(new Partition(Host, item.StartSector, item.SectorCount));
-                        if (MountedPartitions[i] != null)
-                        {
-                            var data = MountedPartitions[i];
-                            part.RootPath = data.RootPath;
-                            part.MountedFS = data;
-                        }
-                        converted.Add(part);
-                        i++;
-                    }
-                }
+                //var converted = new List<ManagedPartition>();
+                //if (GPT.IsGPTPartition(Host))
+                //{
+                //    var gpt = new GPT(Host);
+                //    int i = 0;
+                //    foreach (var item in gpt.Partitions)
+                //    {
+                //        var part = new ManagedPartition(new Partition(Host, item.StartSector, item.SectorCount));
+                //        if (MountedPartitions[i] != null)
+                //        {
+                //            var data = MountedPartitions[i];
+                //            part.RootPath = data.RootPath;
+                //            part.MountedFS = data;
+                //        }
+                //        converted.Add(part);
+                //        i++;
+                //    }
+                //}
+                //else
+                //{
+                //    var mbr = new MBR(Host);
+                //    int i = 0;
+                //    foreach (var item in mbr.Partitions)
+                //    {
+                //        var part = new ManagedPartition(new Partition(Host, item.StartSector, item.SectorCount));
+                //        if (MountedPartitions[i] != null)
+                //        {
+                //            var data = MountedPartitions[i];
+                //            part.RootPath = data.RootPath;
+                //            part.MountedFS = data;
+                //        }
+                //        converted.Add(part);
+                //        i++;
+                //    }
+                //}
 
-                return converted;
+                return parts;
             }
         }
         /// <summary>
         /// List of file systems.
         /// </summary>
-        [Obsolete("use FileSystemManager.RegisteredFileSystems")]
```

Comment:
this change here will been to be reverted

Reply:
the "registeredFileSystems" in the disk.cs was not updated

Reply:
The Change was reveted